packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -67,38 +70,62 @@ export function makeRumPublicApi<C extends RumUserConfiguration>(startRumImpl: S
     })
   }
 
-  const rumPublicApi = makePublicApi({
-    init: monitor((userConfiguration: C) => {
-      if (
-        !checkCookiesAuthorized(buildCookieOptions(userConfiguration)) ||
-        !checkIsNotLocalFile() ||
-        !canInitRum(userConfiguration)
-      ) {
-        return
-      }
-      if (userConfiguration.publicApiKey) {
-        userConfiguration.clientToken = userConfiguration.publicApiKey
+  function initRum(userConfiguration: C) {
+    if (
+      !checkCookiesAuthorized(buildCookieOptions(userConfiguration)) ||
+      !checkIsNotLocalFile() ||
+      !canInitRum(userConfiguration)
+    ) {
+      return
+    }
+    if (userConfiguration.publicApiKey) {
+      userConfiguration.clientToken = userConfiguration.publicApiKey
+    }
+
+    const { configuration, internalMonitoring } = commonInit(userConfiguration, buildEnv)
+    if (!configuration.isEnabled('view-renaming') || !configuration.trackViewsManually) {
+      doStartRum()
+    } else {
+      // drain beforeInitCalls by buffering them until we start RUM
+      // if we get a startView, drain re-buffered calls before continuing to drain beforeInitCalls
+      // in order to ensure that calls are processed in order
+      const beforeInitCalls = bufferApiCalls
+      bufferApiCalls = new BoundedBuffer()
+
+      startViewStrategy = (name) => {
+        doStartRum(name)
       }
+      beforeInitCalls.drain()
```

Comment:
I have a bit of hard time wrapping my head around this, but it seems to handle our use cases nicely with little effort, so great!

Reply:
yes, it handles our case without the need to modify the existing APIs but I felt the need to add some explanation about what is going on 😕 
Are you confortable with it?

Reply:
Yes it LGTM. I can't think of an alternative without introducing a lot more tricky code..